### DISCUSSION TITLE
GitHub action - hosted runner - how to reboot inbetween steps?

### DISCUSSION THREAD

**Discussion by JanneKiiskila** [community/community#53228] [Actions]
### Select Topic Area

Question

### Body

I am trying to automate a snapcraft snap-file testing. However, due to the nature of the snap we must do reboot between `snap install <snapfile>` and before we actually start using/testing it. Is this something that can be done with a GitHub hosted runner?

**Comment by airtower-luna** [comment]
If you reboot the runner during a job, the job will break. What you could do is set up a VM (or maybe a container, if you can make Snap work in it) and test your package inside that.

**Comment by JanneKiiskila** [comment]
Unfortunately I need to run Docker itself on the image, so I think having a docker on top of a docker is probably too much. Time to get a dedicated HW into play and run the tests there with a special local GitHub runner.

**Comment by Tamaraa01x** [comment]
```
``
```
